### PR TITLE
Refactor logger to Ruby standard Logger interface

### PR DESCRIPTION
**Pull request by kkaempf**
Use the standard Ruby Logger class and API for logging

### CODE REVIEW

**Comment by kkaempf** [comment]
This also fixes the config.yml 'debug' key to provide debug levels.
And it adds the ability to log to other files, like stderr.

**Comment by MaximilianMeister** [comment]
@kkaempf cool thanks!!

can you please rebase? I have merged one of the other PRs and pushed the fix for the layout

**Comment by kkaempf** [comment]
Rebase done.

File: lib/nailed/github.rb [hunk 1]
```diff
@@ -96,20 +97,19 @@ def update_pull_states
...
+        Nailed.logger.info("#{__method__}: Checking state of pullrequest #{number} from #{org}/#{repo}")
         if github_pull.state == "closed"
-          Nailed.log("info", "#{__method__}: Deleting closed pullrequest #{number} from #{org}/#{repo}")
+          Nailed.logger.info("#{__method__}: Deleting closed pullrequest #{number} from #{org}/#{repo}")
           db_pull.destroy
         end
       end
     end
 
-    def write_pull_trends(org, repo)
```
Comment (MaximilianMeister) [inline lib/nailed/github.rb]: oops :-)

File: config/config.yml [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ---
-debug: # true for debugging
+debug: debug # fatal, error, warn, info, debug; defaults to 'fatal'
+logfile: stderr # stderr, stdout, or path; defaults to log/nailed.log
```
Comment (MaximilianMeister) [inline config/config.yml]: you forgot to delete your values here i think

**Comment by MaximilianMeister** [comment]
other than that looks good to me! thanks for refactoring that :)

I was always wishing for logs on stdout.
do you think it is worth having stdout as the default logfile? I would defiunitely prefer that. what do you think?

File: config/config.yml (hunk 2 above)
Comment (kkaempf) [inline config/config.yml]: No, both entries are part of the PR.

Previously, you could only log to a file. Now you can log to stdout or stderr.

'debug' was a mess for itself. It was 'debug: true' in the config.yml, code tested for 'Debug' key, and string values (log levels).

File: lib/nailed/github.rb (hunk 1 above)
Comment (kkaempf) [inline lib/nailed/github.rb]: Fixed ;-)

**Comment by kkaempf** [comment]
Just set "logfile: stdout" in the config/config.yml in git. ;-)

File: config/config.yml (hunk 2 above)
Comment (MaximilianMeister) [inline config/config.yml]: maybe you misunderstood me here. in the config.yml we have only the keys and the possible values **commented out**. the keys you added are right, only the values shouldn't be in there, as the user writes those in the file

so `debug:` and `logfile:` should stay and `debug` and `stderr` goes out

**Comment by MaximilianMeister** [comment]
> Just set "logfile: stdout" in the config/config.yml in git. ;-)

yes, i know i can do that now thanks to your PR. i was only wondering to make it implicit. meaning having no `logfile` option set in `config.yml` which would then fall back to stdout as default

File: config/config.yml (hunk 2 above)
Comment (kkaempf) [inline config/config.yml]: Ack, sorry, will move comment characters now.

**Comment by kkaempf** [comment]
`logfile` defaults to stderr, that should largely fulfill your wish ;-)

File: lib/nailed/github.rb [hunk 3]
```diff
@@ -70,18 +71,18 @@ def get_open_pulls
...
                 db_handler = Pullrequest.first_or_create(attributes)
-                Nailed.log("info", "#{__method__}: Created new pullrequest #{pr.repo} ##{pr.number} with #{attributes.inspect}")
+                Nailed.logger.info("#{__method__}: Created new pullrequest #{pr.repo} ##{pr.number} with #{attributes.inspect}")
               end
 
               Nailed.save_state(db_handler) unless defined? db_handler
-              Nailed.log("info", "#{__method__}: Saved #{attributes.inspect}")
+              Nailed.logger.info("#{__method__}: Saved #{attributes.inspect}")
             end unless pulls.empty?
-            write_pull_trends(organization, repo)
```
Comment (MaximilianMeister) [inline lib/nailed/github.rb]: mistake here removing the organization

File: lib/nailed/github.rb [hunk 4]
```diff
@@ -96,34 +97,33 @@ def update_pull_states
...
+          Nailed.logger.info("#{__method__}: Deleting closed pullrequest #{number} from #{org}/#{repo}")
           db_pull.destroy
         end
       end
     end
 
-    def write_pull_trends(org, repo)
-      Nailed.log("info", "#{__method__}: Writing pull trends for #{org}/#{repo}")
-      open = Pullrequest.count(:repository_rname => repo)
+    def write_pull_trends(repository)
```
Comment (MaximilianMeister) [inline lib/nailed/github.rb]: here as well

File: lib/nailed/github.rb [hunk 5]
```diff
@@ -96,34 +97,33 @@ def update_pull_states
...
+      Nailed.logger.info("#{__method__}: Writing pull trends for #{repository}")
+      open = Pullrequest.count(:repository_rname => repository)
       attributes = {:time => Time.new.strftime("%Y-%m-%d %H:%M:%S"),
                     :open => open,
-                    :repository_rname => repo,
-                    :repository_organization_oname => org}
```
Comment (MaximilianMeister) [inline lib/nailed/github.rb]: and this should also stay

**Comment by MaximilianMeister** [comment]
I will merge this now and do the fix right away, thanks again for this! logging to stdout helps a lot in debugging :)

**Pull request merged**